ui/app/clinical/consultation/controllers/addTreatmentController.js

```diff
@@ -27,8 +27,115 @@ angular.module('bahmni.clinical')
+                        $scope.categories.push(resp[i].names[0]);
+                    }
+                    return $scope.categories;
+                });
+            };
+
+            $scope.selectCategoryFromDropdown = function () {
+                $scope.selectedTreatmentLine = "";
+                if ($scope.selectedCategory != undefined) {
+                    if ($scope.selectedCategory.display === "ARV") {
+                        $scope.selectedTreatmentLineCategory = "treatment_line_arv";
+                        $scope.disableTreatmentLine = false;
+                        $scope.disableMedication = false;
+                        $scope.fetchTreatmentLines($scope.selectedTreatmentLineCategory);
+                    } else if ($scope.selectedCategory.display.toUpperCase() === "Anti-Tuberculosis Drugs".toUpperCase() || $scope.selectedCategory.display.toUpperCase() === "Drogas Anti-Tuberculose".toUpperCase()) {
+                        $scope.selectedTreatmentLineCategory = "treatment_line_tb";
+                        $scope.fetchTreatmentLines($scope.selectedTreatmentLineCategory);
+                        $scope.disableTreatmentLine = false;
+                        $scope.disableMedication = false;
+                    } else if ($scope.selectedCategory.display.toUpperCase() === "Other".toUpperCase() || $scope.selectedCategory.display.toUpperCase() === "Outra".toUpperCase()) {
+                        $scope.disableMedication = false;
+                    } else {
+                        $scope.selectedTreatmentLineCategory = "";
+                        $scope.disableTreatmentLine = true;
+                        $scope.disableMedication = false;
+                    }
+                }
+            };
+
+            $scope.fetchTreatmentLines = function (conceptName) {
+                $scope.treatmentLines = [];
+                return conceptSetService.getConcept({
+                    name: conceptName,
+                    v: "custom:(answers:(uuid,name,names))"
+                }, true).then(function (response) {
+                    var resp = response.data.results[0].answers;
+                    for (var i = 0; i < resp.length; i++) {
+                        resp[i].names[0].uuid = resp[i].uuid;
+                        $scope.treatmentLines.push(resp[i].names[0]);
+                    }
+                    return $scope.treatmentLines;
+                });
+            };
+
+            $scope.getDrugsFromTreatmentLineOrCategory = function () {
+                var conceptName = "";
+                if ($scope.selectedTreatmentLine !== undefined && $scope.selectedTreatmentLine !== "") {
+                    conceptName = $scope.selectedTreatmentLine.name;
+                } else if ($scope.selectedCategory !== undefined && $scope.selectedCategory !== "") {
+                    conceptName = $scope.selectedCategory.name;
+                }
+                if (defaultLocale === "en") {
```

Comment:
For sure @nthfloor ! Maybe we can ask for a ticket  to the BA team in order to fix this issues and improve code quality!

Reply:
Agreed @Edgencio, I have held off doing this for the moment since we already have so much going on in prep for the final release - best we focus on closing that out without distractions.

I have a list and I encourage you to also keep a list of the places you see where we need to go back to refactor either as technical debt or in order to contribute back to Bahmni.

After final release we can send this list to Lalitha and start working on them